packages/platform-ios/src/commands/runIOS/index.ts

```diff
@@ -83,11 +86,20 @@ function runIOS(_: Array<string>, ctx: Config, args: FlagsT) {
     );
   }
 
-  const devices = parseIOSDevicesList(
-    child_process.execFileSync('xcrun', ['instruments', '-s'], {
-      encoding: 'utf8',
-    }),
-  );
+  const env = await getEnvironmentInfo(true);
```

Comment:
Running full envinfo is pretty slow. Can we optimize it so that only Xcode info is searched for?

Reply:
envinfo exposes all it's helpers... we could add getXcodeVersion() to envinfo and run only that one helper and maybe even parse (split at '/' as it includes build number) it right there.

Reply:
I think a better way would be to revert all `envinfo` related changes in this PR and just do something like:
```js
    child_process.execSync('xcrun -f xctrace');
```
This will throw when `xctrace` is not available, which is fine.

Something like this:
```js
try {
    // This will throw when `xctrace` is not available. We fallback to `instruments`.
    child_process.execSync('xcrun -f xctrace', {stdio: 'ignore'});
    devices = parseXtraceIOSDevicesList(...);
  } catch (e) {
    devices = parseIOSDevicesList(...);
  }
```
should do the trick.

Reply:
^ with that in mind, we don't need `semver` either.